### PR TITLE
Hero video

### PR DESCRIPTION
**Pull request by davidpmccormick**
## What is this PR trying to achieve?
Adding a video as main media (i.e. to the hero area) on an article page (as a variant in Cardigan).

## What does it look like?
screen shot 2017-03-15 at 18 08 10

### CODE REVIEW

File: server/views/components/main-media/index.njk [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
-{% block figcaption %}
-  {% component 'wobbly-edge', {background: 'cream'} %}
-  {% if model.caption %}
-    <div class="row row--cream">
-      <div class="container">
-        <div class="grid">
-          <div class="{{ {m:6, xl:10, shiftM:1, shiftXl:1} | gridClasses }}">
-            {{ super() }}
+{% if model.type === 'picture' %}
+  {% extends 'components/captioned-image/index.njk' %}
```
Comment (jamesgorrie) [inline server/views/components/main-media/index.njk]: Nice, didn't know you could have conditional `extends`.

File: server/views/components/main-media/index.njk [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+          {% componentV2 'iframed-video', {
+            embedUrl: model.embedUrl
+          } %}
+        </div>
+      </div>
     </div>
-  {% endif %}
-{% endblock %}
+  </div>
+  {% component 'wobbly-edge', {background: 'cream'} %}
```
Comment (jamesgorrie) [inline server/views/components/main-media/index.njk]: We have a wobbly edge on the video?
Isn't that going to be distracting while watching?

File: server/views/components/page-description/index.njk [hunk 3]
```diff
@@ -1,27 +1,30 @@
-<header class="page-description {% for modifier in modifiers %}page-description--{{ modifier }} {% endfor %}">
-  <div class="row page-description__row {{ 'row--lead row--lead--l' if lead }}">
+<header class="page-description {{ 'page-description' | componentClasses(modifiers) }}">
+  <div class="row page-description__row {{ 'row--lead row--lead--l' if model.lead }}">
     <div class="container page-description__container">
       <div class="grid">
-        <div class="grid__cell grid__cell--s4 grid__cell--m6 grid__cell--shift-m1 {{ 'grid__cell--shift-l1 grid__cell--l11 grid__cell--xl10' if lead else 'grid__cell--l12 grid__cell--xl11' }}">
-          <h1 class="page-description__title">{{ title }}</h1>
+        <div class="grid__cell grid__cell--s4 grid__cell--m6 grid__cell--shift-m1 {{ 'grid__cell--shift-l1 grid__cell--l11 grid__cell--xl10' if model.lead else 'grid__cell--l12 grid__cell--xl11' }}">
```
Comment (jamesgorrie) [inline server/views/components/page-description/index.njk]: It'd be nice to wrap these - but that's for another PR.

File: server/views/pages/article.njk [hunk 4]
```diff
@@ -7,16 +7,16 @@
 {% endblock %}
 
 {% block body %}
-{% component 'page-description', { title: article.headline, meta: { type: 'date', value: article.datePublished | formatDate }, modifiers: ['a'], lead: true } %}
+
+{% componentV2 'page-description', {title: article.headline, meta: {type: 'date', value: article.datePublished | formatDate}, lead: true}, {'a': true}, {icon: article.contentType | getIconForContentType} %}
```
Comment (jamesgorrie) [inline server/views/pages/article.njk]: Could we put the format date into the component?

File: server/views/pages/article.njk [hunk 5]
```diff
@@ -7,16 +7,16 @@
...
 
 {% if article.series[0] and article.series[0].commissionedLength %}
   {% set asyncContentEndpoint = '/series-nav/' + article.series[0].url + '?current=' + article.url %}
   {% component 'async', { endpoint: asyncContentEndpoint, placeholderClassname: 'series-nav-placeholder' } %}
 {% endif %}
 
-
 {% for media in article.mainMedia %}
-  {% if (media.contentUrl) %}
+  {% if (media.contentUrl) or (media.embedUrl) %}
```
Comment (jamesgorrie) [inline server/views/pages/article.njk]: This is a hard to decipher conditional - but admittedly it was me that put the first one in.
What instances do we have where we have main media without these?

File: server/views/components/main-media/index.njk (hunk 2 above)
Comment (davidpmccormick) [inline server/views/components/main-media/index.njk]: As with the hero image wobbly edge, it only wobbles on scroll (and doesn't cut into the video) – think we should be OK.

File: server/views/pages/article.njk (hunk 5 above)
Comment (davidpmccormick) [inline server/views/pages/article.njk]: I have to confess to being lazy and not figuring that out – I thought it was safer to go this way round than to remove the conditional altogether.

File: server/views/pages/article.njk (hunk 4 above)
Comment (davidpmccormick) [inline server/views/pages/article.njk]: Surely. Will update.

File: server/views/components/main-media/index.njk (hunk 1 above)
Comment (davidpmccormick) [inline server/views/components/main-media/index.njk]: Yeah, I fully expected it to blow up, too.

File: server/views/pages/article.njk (hunk 5 above)
Comment (jamesgorrie) [inline server/views/pages/article.njk]: Let's scrap it.

**Pull request merged**